Modules/StudyProgramme/classes/class.ilObjStudyProgrammeAccess.php

```diff
@@ -14,7 +16,25 @@
  * @author: Richard Klees <richard.klees@concepts-and-training.de>
  *
  */
-class ilObjStudyProgrammeAccess extends ilObjectAccess {
+class ilObjStudyProgrammeAccess extends ilObjectAccess implements ilConditionHandling{
+
+	/**
+	 * @var ilStudyProgrammeUserProgressDB
+	 */
+	protected $sp_user_progress_db;
+
+	/**
+	* Get a (cached) instance of ilStudyProgrammeUserProgressDB
+	*
+	* @return ilStudyProgrammeUserProgressDB
+	*/
+	private function getStudyProgrammeUserProgressDB() {
```

Comment:
This needs to become static, because we should not call it from `checkCondition` otherwise. Also you do not need to cache the DB, `ilObjStudyProgramme` already does that. I just realised that this also means that this complete method could just go away an be replaced by the according method on `ilObjStudyProgramme`.